app/src/main/java/io/keen/client/android/example/MyActivity.java

```diff
@@ -56,16 +61,63 @@ protected void onPause() {
     }
 
     public void handleClick(View view) {
+        Log.i(this.getLocalClassName(), "handling click on view " + view.getId());
         switch (view.getId()) {
             case R.id.send_event_button:
                 // Create an event to upload to Keen.
-                Map<String, Object> event = new HashMap<String, Object>();
+                Map<String, Object> event = new HashMap<>();
                 event.put("click-number", clickNumber++);
 
                 // Add it to the "purchases" collection in your Keen Project.
-                KeenClient.client().queueEvent("android-sample-button-clicks", event);
+                KeenClient.client().queueEvent(getCollection(), event);
+                break;
+            case R.id.query_button:
+                new KeenQueryAsyncTask().execute();
                 break;
         }
     }
 
+    private KeenProject getProject() {
+        if (project == null) {
+            // Get the project ID and write key from string resources, then create a project and set
+            // it as the default for the client.
+            String projectId = getString(R.string.keen_project_id);
```

Comment:
wow, that's some ugly code. guess it has to do with how android injects resources?